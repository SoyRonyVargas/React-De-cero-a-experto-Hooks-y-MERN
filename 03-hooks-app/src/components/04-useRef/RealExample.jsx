import React, { useState } from 'react'
import MultipleHooks from '../03-custom-hooks/MultipleHooks'

const RealExample = () => {
    
    const [ show , setShow ] = useState(false)
    
    return (
        <div>
            
            <h1>RealExampleRef</h1>
            
            <hr />

            <button onClick={ () => setShow(!show) } className="btn btn-primary">
                {
                    show ? 'Show' : 'Hide'
                }
            </button>

            {
                show &&
                <MultipleHooks/>
            }

        </div>
    )
}

export default RealExample
