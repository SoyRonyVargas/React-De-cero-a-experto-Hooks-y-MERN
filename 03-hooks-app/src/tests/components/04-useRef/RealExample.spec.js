import RealExample from '../../../components/04-useRef/RealExample'
import { shallow } from 'enzyme'

describe('Probando el componente RealExample', () => { 
    
    test('deberia mostrarse correctamente', () => { 
        
        const wrapper = shallow(<RealExample/>)

        expect(wrapper).toMatchSnapshot()

    })

    test('No deberia mostrar el componente <MultipleHooks/>', () => { 

        const wrapper = shallow(<RealExample/>)

        expect(wrapper.find("#container").exists()).toBe(false)

    })
    
})