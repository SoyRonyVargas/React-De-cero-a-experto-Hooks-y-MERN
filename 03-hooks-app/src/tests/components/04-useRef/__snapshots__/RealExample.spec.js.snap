// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente RealExample deberia mostrarse correctamente 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Hide
  </button>
</div>
`;
